### PR TITLE
Reset, don't remove sub-resource properties to avoid wrong diffs

### LINKED ISSUE #3049: Refresh for VirtualNetwork with external subnets still shows subnets
I'm running [`vnet-subnets-resolution`](https://github.com/pulumi/pulumi-azure-native/tree/master/examples/vnet-subnets-resolution) example with the following flow.

1. `pulumi up` and deploy all resources, all goes well.
2. `pulumi refresh` and expect no meaningful changes, at least not in `subnets` property of the `external` network. However, I get this: 
image
image

3. Run `pulumi up` again, it shouldn't try deleting the subnets but it does suggest doing so:
image

Expected result: no refresh or update of `subnets`.

I think this regressed since https://github.com/pulumi/pulumi-azure-native/pull/2755, most likely in https://github.com/pulumi/pulumi-azure-native/pull/2950. A quick debugging session shows that the `removeUnsetSubResourceProperties` invocation is trying to delete paths `[properties subnets]` - note the top-level `properties` here. However, it can't delete it from the SDK shape, because `properties` are flattened here, so it's probably operating on incompatible data structures.

If you are interested, I have an automated test for this - will open PRs for it soon.

Azure Native version: 2.27.0.
CLI version: 1.103.1.

### PR DESCRIPTION
**Pull request by thomas11**
This PR fixes a defect introduced by #2950. When handling sub-resource properties that may be maintained as stand-alone resources, we need to overwrite them in several code paths. Before, we set them to their default value `[]` on Create, but removed them entirely on Read. This causes unnecessary diffs on Refresh. This PR standardizes the handling on resetting to the default value `[]`.

Resolves #3049

### CODE REVIEW

**Comment by mikhailshilkov** [comment]
My PR here has the tests that caught the original bug that I filed: https://github.com/pulumi/pulumi-azure-native/pull/3056

File: provider/pkg/provider/provider_test.go [hunk 1]
```diff
@@ -458,20 +430,26 @@ func TestRemoveUnsetSubResourceProperties(t *testing.T) {
...
+					"a policy",
+				},
+			},
+		}
+		actual := provider.resetUnsetSubResourceProperties(ctx, "urn", sdkResponse, oldInputs, res)
+		expected := map[string]any{
 			"properties": map[string]any{
 				"accessPolicies": []any{},
 			},
 		}
```
Comment (danielrbradley) [inline provider/pkg/provider/provider_test.go:452]: Looks like this is the actual core of the change right? We're now expecting the access policies to still be there but be set to an empty value?

File: provider/pkg/provider/provider_test.go [hunk 2]
```diff
@@ -553,33 +532,53 @@ func TestSetUnsetSubresourcePropertiesToDefaults(t *testing.T) {
 
 	t.Run("unchanged", func(t *testing.T) {
 		body := map[string]any{
-			"properties": map[string]any{
-				"accessPolicies": []any{},
+			"container": map[string]any{
+				"properties": map[string]any{
+					"accessPolicies": []any{},
+				},
```
Comment (danielrbradley) [inline provider/pkg/provider/provider_test.go:538]: What's the reason for adding the container? Were we missing an edge case?

**Review by danielrbradley — COMMENTED** [review]
I _think_ this is probably good, though it's still tricky to reason about how wide the impact will be.

I feel like we need a succinct description of this feature written down somewhere - like a short spec which describes the behaviour. Perhaps this could go into the contributing guide where we discuss things like the versioning mechanisms or perhaps a block alongside the tests could be easier to find? 

My gut feel is that the tests are still a little patchy and are missing a few edge cases - though I haven't gathered proper evidence for this.

File: provider/pkg/provider/provider_test.go (hunk 2 above)
Comment (thomas11) [reply provider/pkg/provider/provider_test.go:538]: Just expanding test coverage. I added a test case below for the SDK-shape case, without the container, so both are covered.

File: provider/pkg/provider/provider_test.go (hunk 1 above)
Comment (thomas11) [reply provider/pkg/provider/provider_test.go:452]: Correct. Because on `Create`, we set such properties to their default value, which is `[]`, so we want it here, too, to avoid diffs.

**Review by danielrbradley — APPROVED** [review]
